### PR TITLE
Update for support of Python 3.11 in NumPy 1.23.2

### PR DESCRIPTION
**Pull request by charris**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -66,5 +67,4 @@ install_requires =
     # we specify an unpinned Numpy which allows source distributions
     # to be used and allows wheels to be used as soon as they
     # become available.
-    numpy; python_version>='3.11'
```
Comment (rgommers) [inline setup.cfg:69]: This isn't quite right, instead the version number should be changed to 3.12 here

File: setup.cfg [hunk 2]
```diff
@@ -57,6 +57,7 @@ install_requires =
     # Note that 1.21.3 was the first version with a complete set of 3.10 wheels,
     # however macOS was broken and it's safe to build against 1.21.6 on all platforms (see gh-28 and gh-45)
     numpy==1.21.6; python_version=='3.10' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'
+    numpy==1.23.2; python_version>='3.11' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'
```
Comment (rgommers) [inline setup.cfg:60]: This should be `=='3.11'

**Review by rgommers — DISMISSED** [review]
Thanks Chuck! A couple of minor changes needed

**Comment by charris** [comment]
Done.

**Comment by rgommers** [comment]
I spotted another inconsistency with `loongarch`, which should no longer be special-cased. Pushed a fix, and a test addition. Could you do a sanity check @charris?

**Comment by charris** [comment]
Can't say my review is worth much, but LGTM. Is it the case that Python 3.11 supports loongarch64?

**Comment by rgommers** [comment]
> Is it the case that Python 3.11 supports loongarch64?

Nothing specific is needed there for new Python versions. `loongarch64` is simply a CPU architecture that exists, and needed special-casing because it needed a few patches in NumPy (and probably Python and elsewhere too). Now that those have landed, it's just one of many architectures supported in Debian.

**Comment by rgommers** [comment]
In it goes, thanks @charris. I'll make a release soon.

File: setup.cfg [hunk 3]
```diff
@@ -56,8 +56,10 @@ install_requires =
     numpy==1.19.3; python_version=='3.9' and platform_machine not in 'arm64|loongarch64' and platform_python_implementation != 'PyPy'
     # Note that 1.21.3 was the first version with a complete set of 3.10 wheels,
     # however macOS was broken and it's safe to build against 1.21.6 on all platforms (see gh-28 and gh-45)
-    numpy==1.21.6; python_version=='3.10' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'
+    numpy==1.23.2; python_version=='3.10' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'
```
Comment (rgommers) [inline setup.cfg:59]: Grr, this change was unintended, reverting that

Comment (rgommers) [reply setup.cfg:59]: Force-pushed commit 0688c4bbf25322194e249ec1f8072961f8fe02e3 to repair this. Done within minutes, so should be okay.

**Comment by lpsinger** [comment]
Would you please do a release to get this on PyPI? This will help downstream packages that depend on Numpy start uploading Python 3.11 wheels to PyPI.

**Comment by rgommers** [comment]
> Would you please do a release to get this on PyPI? This will help downstream packages that depend on Numpy start uploading Python 3.11 wheels to PyPI.

I did a release straight after merging it: https://pypi.org/project/oldest-supported-numpy/2022.8.16/. Is that not working for you?

**Comment by lpsinger** [comment]
You're right. I was confused by the output of my CI pipeline. Thanks!

**Pull request merged**